src/VirtualClient/VirtualClient.Actions/ScriptExecutor/ScriptExecutor.cs

```diff
@@ -0,0 +1,268 @@
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.CommandLine));
+            }
+        }
+
+        /// <summary>
+        /// The relative Script Path to be used to initiate the script
+        /// </summary>
+        public string ScriptPath
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.ScriptPath));
+            }
+        }
+
+        /// <summary>
+        /// The Regex Based, semi-colon separated relative log file/Folder Paths
+        /// </summary>
+        public string LogPaths
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.LogPaths));
+            }
+        }
+
+        /// <summary>
+        /// The ToolName for better logging and metadata
+        /// </summary>
+        public string ToolName
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.ToolName));
+            }
+        }
+
+        /// <summary>
+        /// The full path to the script executable.
+        /// </summary>
+        public string ExecutablePath { get; set; }
+
+        /// <summary>
+        /// The path to the workload package.
+        /// </summary>
+        protected DependencyPath WorkloadPackage { get; set; }
+
+        /// <summary>
+        /// Initializes the environment for execution of the provided script.
+        /// </summary>
+        protected override async Task InitializeAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            await this.EvaluateParametersAsync(cancellationToken);
+
+            this.WorkloadPackage = await this.GetPlatformSpecificPackageAsync(this.PackageName, cancellationToken);
+
+            this.ExecutablePath = this.Combine(this.WorkloadPackage.Path, this.ScriptPath);
+
+            await this.systemManagement.MakeFileExecutableAsync(this.ExecutablePath, this.Platform, cancellationToken);
+
+            if (!this.fileSystem.File.Exists(this.ExecutablePath))
+            {
+                throw new DependencyException(
+                    $"The expected workload binary/executable was not found in the '{this.PackageName}' package. The script cannot be executed " +
+                    $"successfully without this binary/executable. Check that the workload package was installed successfully and that the executable " +
+                    $"exists in the path expected '{this.ExecutablePath}'.",
+                    ErrorReason.DependencyNotFound);
+            }
+        }
+
+        /// <summary>
+        /// Executes the workload.
+        /// </summary>
+        protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            using (BackgroundOperations profiling = BackgroundOperations.BeginProfiling(this, cancellationToken))
```

Comment:
Let's add just 1 or 2 comments about how logs are uploaded and then moved?

Reply:
Added in summary of the CaptureLogsAsync method